packages/@aws-cdk/cx-api/lib/features.ts

```diff
@@ -56,6 +56,15 @@ export const STACK_RELATIVE_EXPORTS_CONTEXT = '@aws-cdk/core:stackRelativeExport
  */
 export const DOCKER_IGNORE_SUPPORT = '@aws-cdk/aws-ecr-assets:dockerIgnoreSupport';
 
+/**
+ * Secret.secretName for an "owned" secret will attempt to parse the secretName from the ARN,
+ * rather than the default full resource name, which includes the SecretsManager suffix.
+ *
+ * If this flag is not set, Secret.secretName will include the SecretsManager suffix, which cannot be directly
+ * used by SecretsManager.DescribeSecret, and must be parsed by the user first (e.g., Fn:Join, Fn:Select, Fn:Split).
+ */
+export const SECRETS_MANAGER_PARSE_OWNED_SECRET_NAME = '@aws-cdk/aws-secretsmanager:parseOwnedSecretName';
```

Comment:
Come to think about this a bit more this might be a bit tricky to introduce under a global feature flag. 

Consider an app that enables this feature which uses a library that was developed with this feature disabled. Now the library will effectively get a different behavior then it originally was tested with.

I recon it's unlikely to happen, especially given 2.0 is not too far out and we'll clean this up... but thought I'd bring it up.

Reply:
It's a valid point. That being said, I think there are four mitigations.
1. As you said, 2.0 is not far out.
2. This functionality (owned secret names) has only been out for three releases, limiting the timeframe for someone to have built a library based on this.
3. The existing functionality really is only marginally useful; any use case I'm aware of would likely need to be manually using Split/Select to make use of the name, so I think more users are avoiding it rather than embracing it (for owned secrets).

That being said, one thing I *can* do is add something to the README or secretName docs to advertise this feature flag so a user is less likely to get the wrong behaviour.